### ISSUE TITLE
When switching mobile data to WiFi or back, the internet freezes. Android 14.

### ISSUE THREAD

**Issue by OBrein** [xchacha20-poly1305/husi#40]
### Version

```text
0.5.0-rc1
```

### Description

When switching mobile data to WiFi or back, the internet freezes.  The same effect when switching SIM cards. For normal operation, you need to turn it off and on again every time you change the network. Android 14, Galaxy Fold 5.

### Reproduction

Just change network from SIM1 to SIM2 or WIFI

### Logs

```shell
ERROR[0022] router: missing default interface
ERROR[0023] [4244340092 8ms] inbound/tun[tun-in]: dial tcp MYIP:PORT: connect: network is unreachable
```

### Integrity requirements

- [X] I confirm that my problem is the bug of husi, but not the problem of my own network environment or my server.
- [X] I confirm that I have provided the simplest configuration that can be used to reproduce the error I reported.
- [X] I confirm that I have provided the complete configuration files and logs, rather than just providing parts I think are useful out of confidence in my own intelligence.

**Comment by OBrein** [comment]
No connections to 30 minutes. May be samsung bug. >_<
Please watch the video. There is no network until I reconnect husi.

https://github.com/xchacha20-poly1305/husi/assets/54883042/253e1325-5209-4b1b-8c1b-c22fbe5dd2f8

**Comment by OBrein** [comment]
That's what I mean. The smartphone itself loses its internet connection before the husi reboot. It reconnects normally without husi. It reconnects normally via Nekobox (1.8.0 from actions). Android itself does not reset tun. It is unclear why. May be should I send the full log?

**Comment by OBrein** [comment]
> Just provide the video of switching mobile data to WIFI.
There is no wifi in my location right now. I'll make a video tonight.  🤝

**Comment by OBrein** [comment]
Switching from wifi to mobile data. There is no network (even in the notification bar) before manually restarting husi.
https://github.com/xchacha20-poly1305/husi/assets/54883042/a838bc89-1bd1-44e3-bbbd-b18d8d92702f

**Comment by OBrein** [comment]
My log file
P.S. sing-box directly from command line (termux app) - netlink error
FATAL[0000] create service: parse route options: netlink socket in Android is banned by Google, use the root or system (ADB) user to run sing-box, or switch to the sing-box Adnroid graphical interface client
[husi 8349831873429245672.log](https://github.com/xchacha20-poly1305/husi/files/14400854/husi.8349831873429245672.log)

**Comment by xchacha20-poly1305** [comment]
> This bug is introduced in [de7be95](https://github.com/xchacha20-poly1305/husi/commit/de7be957762a911a2054d646a3620172acc9c0f7).
> 
> Here is a patch to "fix" it.

蓬荜生辉！

**Comment by OBrein** [comment]
Fix in version 0.6.7, thanks 🙏
